pkg/util/serviceaccount.go

```diff
@@ -83,7 +83,25 @@ func WaitForServiceAccountSecretCreation(client kubeclient.Interface, asObj *cor
 			}
 			return false, fmt.Errorf("failed to retrieve service account(%s/%s) from cluster, err: %v", asObj.Namespace, asObj.Name, err)
 		}
-		clusterSecret, err = GetTargetSecret(client, serviceAccount.Secrets, corev1.SecretTypeServiceAccountToken, asObj.Namespace)
+		if len(serviceAccount.Secrets) == 0 {
```

Comment:
Is it possible that the secret of the serviceAccount is being created but has not been created successfully?

Reply:
Good point.

Reply:
For kubernetes clusters with versions smaller than v1.22, no token will be generated when a secret is created for ServiceAccount.

Maybe we need to wait a while for the creation of `serviceAccount secret token`, if timeout, we can create the secret.